packages/@aws-cdk/aws-lambda/lib/function.ts

```diff
@@ -487,11 +489,12 @@ export class Function extends FunctionBase {
 
     // Log retention
     if (props.logRetention) {
-      new LogRetention(this, 'LogRetention', {
+      const logretention = new LogRetention(this, 'LogRetention', {
         logGroupName: `/aws/lambda/${this.functionName}`,
         retention: props.logRetention,
         role: props.logRetentionRole
       });
+      this._logGroup = logs.LogGroup.fromLogGroupArn(this, `${this.node.id}-LogGroup`, logretention.logGroupArn);
```

Comment:
Why do we need the `${this.node.id}-` prefix here? Isn't `LogGroup` unique within the construct at this point?

Reply:
Yes, it is. You're right, I can probably drop the prefix.